### ISSUE TITLE
Bun.file().stream() seems to repeat the same chunk over and over

### ISSUE THREAD

**Issue by paperclover** [oven-sh/bun#1101]
### Version

0.1.9

### What steps will reproduce the bug?

I don't have time right now to get a complete and isolated bug reproduction. I was writing a single script for processing a 2.5gb text file and that code is [here](https://github.com/paperdave/stable-diffusion-sqlite).

but the idea is using Bun.stream on a large file, you'll see the first chunk repeat.

```ts
const reader = Bun.file("ImageLinks.txt").stream().getReader();
console.log((await reader.read()).value.slice(0, 10));
console.log((await reader.read()).value.slice(0, 10));
console.log((await reader.read()).value.slice(0, 10));
console.log((await reader.read()).value.slice(0, 10));
// ...
```

### How often does it reproduce? Is there a required condition?

every time

### What is the expected behavior?

iterate over every chunk of the file not just the first one.

### What do you see instead?

```js
~/test$ bun convert-bun.ts 
Uint8Array(10) [ 84, 111, 116, 97, 108, 58, 32, 57, 53, 51 ]
Uint8Array(10) [ 84, 111, 116, 97, 108, 58, 32, 57, 53, 51 ]
Uint8Array(10) [ 84, 111, 116, 97, 108, 58, 32, 57, 53, 51 ]
Uint8Array(10) [ 84, 111, 116, 97, 108, 58, 32, 57, 53, 51 ]
Uint8Array(10) [ 84, 111, 116, 97, 108, 58, 32, 57, 53, 51 ]
Uint8Array(10) [ 84, 111, 116, 97, 108, 58, 32, 57, 53, 51 ]
Uint8Array(10) [ 84, 111, 116, 97, 108, 58, 32, 57, 53, 51 ]
Uint8Array(10) [ 84, 111, 116, 97, 108, 58, 32, 57, 53, 51 ]
Uint8Array(10) [ 84, 111, 116, 97, 108, 58, 32, 57, 53, 51 ]
Uint8Array(10) [ 84, 111, 116, 97, 108, 58, 32, 57, 53, 51 ]
Uint8Array(10) [ 84, 111, 116, 97, 108, 58, 32, 57, 53, 51 ]
Uint8Array(10) [ 84, 111, 116, 97, 108, 58, 32, 57, 53, 51 ]
```

### Additional information

this is not just every chunk being similar; we get completely differnet database results where its missing most of the data due to this bug. the node implementation works but i wanted to use bun, which would lead to around a 3x time improvement because io and sqlite bindings are really great (good job on that).

**Comment by NathanBeddoeWebDev** [comment]
The slice method shouldn't mutate the value you're getting from the reader. I'm not well versed on the API for streams in Bun, so I can't offer up an alternative method (maybe someone else can).

This is just based on my knowledge of the slice method from other APIs
https://nodejs.org/api/buffer.html#blobslicestart-end-type - Node.js doesn't alter buffers on slice either.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice#description - Slicing strings and arrays in javascript doesn't mutate the original value either.

**Comment by paperclover** [comment]
> The slice method shouldn't mutate the value you're getting from the reader. I'm not well versed on the API for streams in Bun, so I can't offer up an alternative method (maybe someone else can).
> 
> This is just based on my knowledge of the slice method from other APIs https://nodejs.org/api/buffer.html#blobslicestart-end-type - Node.js doesn't alter buffers on slice either. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice#description - Slicing strings and arrays in javascript doesn't mutate the original value either.

im just using slice to easily illustrate the arrays are the same, as the actual arrays are much larger. the issue is the fact that **the data from `.read()` remains the same**

**Comment by Electroid** [comment]
It looks like this has been fixed, tested using `0.2.2` on MacOS Silicon.

```js
const reader = Bun.file("cat.jpg").stream().getReader(); // use any file here

console.log(await reader.read()); // { value: Uint8Array(992514) [...], done: false }
console.log(await reader.read()); // { value: undefined, done: true}
console.log(await reader.read()); // { value: undefined, done: true}
```

If you can still reproduce this, please comment and we'll reopen the issue.

**Comment by paperclover** [comment]
can you try this on a larger file? on 0.2.2 with a file that spans two or more chunks, theyre all the same data. in the example i used /dev/urandom which should be continuous random data. also, bun hangs here since the file is never fully read, but i think it should exit in this case.

![image](https://user-images.githubusercontent.com/24465214/200217602-de0f02cd-71a1-4f1e-8255-f1c699f67729.png)

an interesting finding when testing this was using bun.file on /dev/urandom itself and not a real file causes the desired behavior. even the hanging thing is fixed here
![image](https://user-images.githubusercontent.com/24465214/200217905-b0bf27f0-53b6-4b70-ac1d-c8ae6edfbcfc.png)

running this all on `Linux 6.0.7-arch1-1` on bun canary as of the time of this comment.

**Comment by Jarred-Sumner** [comment]
I think I see what’s happening 

macOS has different code for this 

We are not increasing the position offset for the file when we should be

**Comment by Electroid** [comment]
This should be fixed as of Bun [v0.3.0](https://bun.sh/blog/bun-v0.3.0) 🎉
